File: rest/src/main/java/com/tugab/adspartners/service/SubscriptionService.java
package com.tugab.adspartners.service;

import com.tugab.adspartners.domain.entities.Company;
import com.tugab.adspartners.domain.entities.Youtuber;
import com.tugab.adspartners.domain.models.binding.subscription.SubscriberStatusBindingModel;
import com.tugab.adspartners.domain.models.response.subscription.SubscriptionInfoResponse;
import org.springframework.http.ResponseEntity;

import java.util.List;

public interface SubscriptionService {

    public ResponseEntity<List<SubscriptionInfoResponse>> getList(Company company);

    public ResponseEntity<?> subscribe(Youtuber youtuber, Long companyId);

    public ResponseEntity<?> updateStatus(SubscriberStatusBindingModel subscriberStatusBindingModel);

    public ResponseEntity<?> unsubscribe(Youtuber youtuber, Long companyId);

    public ResponseEntity<Boolean> checkSubscription(Long youtuberId, Long companyId);
}

File: ui/src/services/subscriptionService.js
import requester from './requester';

export const getSubscribers = async () => {
    return await requester('/subscription/list', 'GET', true);
};

export const subscribe = async (companyId) => {
    return await requester(`/subscription/subscribe/${companyId}`, 'POST', true);
};

export const updateSubscriberStatus = async (youtuberId, newStatus) => {
    return await requester(`/subscription`, 'PATCH', true, { 
        isBlocked: newStatus,
        youtuberId 
    });
};

export const unsubscribe = async (companyId) => {
    return await requester(`/subscription/unsubscribe/${companyId}`, 'DELETE', true);
};

export const checkSubscription = async (companyId) => {
    return await requester(`/subscription/check/${companyId}`, 'GET', true);
};
